org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingManager.java

```diff
@@ -309,17 +309,37 @@ public void install(JavaSourceViewer sourceViewer, IColorManager colorManager, I
 		install(null, sourceViewer, colorManager, preferenceStore);
 	}
 
+	/**
+	 * Create semantic highlighting presenter instance to use.
+	 *
+	 * @return semantic highlighting presenter to use
+	 * @since 3.28
```

Comment:
Regarding unnecessary `@since` tags on "non API" code ... I considered new non-private methods as API code - API of the classes usable from potential extensions (subclasses). If this reasoning is too strict and thus wrong here, let me know and I'll remove remaining tags.

Reply:
"Since" tags are used by API tooling to check for API breakage between releases, and so make only sense on real API code. Put on internal code they only conffuse. So please remove them.